src/models/peer.rs

```diff
@@ -349,23 +340,12 @@ impl PeerStanding {
         self.standing.is_negative()
     }
 
-    /// Should only be used if peer was expected to be found in the peer map
-    /// but, for some reason, was not there. Please only use this function for
-    /// that purpose.
-    pub fn new_on_no_standing_found(peer_tolerance: u16) -> Self {
-        assert!(
-            peer_tolerance > 0,
-            " peer tolerance must be greater than zero"
-        );
-        Self {
-            standing: NegativePeerSanction::NoStandingFoundMaybeCrash.severity(),
-            latest_punishment: Some((
-                NegativePeerSanction::NoStandingFoundMaybeCrash,
-                SystemTime::now(),
-            )),
-            latest_reward: None,
-            peer_tolerance: peer_tolerance.into(),
-        }
+    pub(crate) fn is_bad(&self) -> bool {
+        self.standing <= self.peer_tolerance
+    }
+
+    pub(crate) fn is_good(&self) -> bool {
+        !self.is_bad()
```

Comment:
Something smells fishy ....

 - `self.standing` can be positive or negative; it starts at 0.
 - Whenever it is updated, it is clamped to `[-self.peer_tolerance, self.peer_tolerance]`.
 - The lower bound is also sticky: once you reach it you are stuck there.
 - Peers that reach the lower bound are banned.

These convenience functions
 - label all peers as "bad"
 - resulting in them being treated as banned.

Reply:
Probably a missing `-` in front of `self.peer_tolerance`. Point of this PR is: get it right once, get it right all the time.

Reply:
> Probably a missing `-` in front of `self.peer_tolerance`.

I did not realize a fix could be so simple. O_o